### ISSUE TITLE
[Bug Report] listed datasets are unavailable for a given version - why list them?

### ISSUE THREAD

**Issue by vmoens** [Farama-Foundation/Minari#200]
This is more a question than a bug report.
In torchrl we test a random set of datasets extracted from `list_remote_datasets`
Since the last release, this test has been broken: https://github.com/pytorch/rl/actions/runs/8649362912/job/23715408536

The reason is that a dataset picked from the list has the [right major version](https://github.com/pytorch/rl/blob/0d007483317b9b1f7298082f1d40401b971220c3/test/test_libs.py#L2560-L2566) (0.4) but the wrong minor, and isn't available for this minor anymore.

My questions are:
- If I want to pick up datasets that work with my version of Minari, how can I do this (and not pick datasets that don't)?
- Why keeping in the list datasets that don't work with my version of minary? If it's for backward referencing, maybe a flag in `list_remote_datasets` to display non-compatible datasets would be useful (I'd expect people to be surprised to see datasets they can't work with in the list except if they ask for them)
- I might be wrong but it seems to me that the dataset that is causing our error (pointmaze-large-dense-v1) has changed in between two minor versions. IMO minor versions should not introduce any change of API unless a bug is solved on the user side. In other words, whatever worked in one major should work in all versions of that major (I understand not all libs abide by this rule but ideally I think it should be the case)

**Comment by alexdavey** [comment]
Hi @vmoens, a new version of the `pointmaze` datasets was recently uploaded. This new upload should not break anything for the old datasets, but I believe there is a bug with the dataset download code which prevents it from loading older dataset versions.

As a temporary fix, could you see if first force downloading the dataset works on your end? E.g.:
```
minari.download_dataset("pointmaze-large-dense-v1", force_download=True)
dataset = minari.load_dataset("pointmaze-large-dense-v1")
```

**Comment by vmoens** [comment]
Thanks @alexdavey for the quick response
I guess my point here is that I don't want to force torchrl's CI to do unorthodox things such as downloading datasets that should not work with the release being used. If you look at the code I'm using above, we get a bunch of dataset names and download them iteratively, checking that they can be used within torchrl. 
What I see here is that there's a dataset that I'm told that is available for 0.4.3 but it doesn't work for 0.4.3.
How can I get a list of the available datasets in 0.4.3 that work with 0.4.3 (without forcing anything)?

**Comment by younik** [comment]
> Thanks @alexdavey for the quick response
> I guess my point here is that I don't want to force torchrl's CI to do unorthodox things such as downloading datasets that should not work with the release being used. If you look at the code I'm using above, we get a bunch of dataset names and download them iteratively, checking that they can be used within torchrl.
> What I see here is that there's a dataset that I'm told that is available for 0.4.3 but it doesn't work for 0.4.3.
> How can I get a list of the available datasets in 0.4.3 that work with 0.4.3 (without forcing anything)?

Yes, all the datasets listed by default with `list_remote_datasets` are compatible with the current version.

The error you see is misleading and a bug: it should only warn you that the version you are downloading is not the latest version.

@alexdavey We also have two different behaviors for `list remote` with CLI and Python code. With CLI we only show the latest version, while with Python code all the compatible versions. I believe we should unify the behaviors.
